hydrolib/core/io/structure/models.py

```diff
@@ -0,0 +1,129 @@
+from enum import Enum
+from hydrolib.core.io.ini.models import IniBasedModel
+from pathlib import Path
+from pydantic import validator, Field
+from typing import List, Literal, Optional, Union
+
+
+# TODO: handle comment blocks
+# TODO: handle duplicate keys
+class Structure(IniBasedModel):
+    # TODO: would we want to load this from something externally and generate these automatically
+    class Comments(IniBasedModel.Comments):
+        id: Optional[str] = "Unique structure id (max. 256 characters)."
+        name: Optional[str] = "Given name in the user interface."
+        branch_id: Optional[str] = Field(
+            "Branch on which the structure is located.", alias="branchId"
+        )
+        chainage: Optional[str] = "Chainage on the branch (m)."
+
+        n_coordinates: Optional[str] = Field(
+            "Number of values in xCoordinates and yCoordinates", alias="numCoordinates"
+        )
+        x_coordinates: Optional[str] = Field(
+            "x-coordinates of the location of the structure. (number of values = numCoordinates)",
+            alias="xCoordinates",
+        )
+        y_coordinates: Optional[str] = Field(
+            "y-coordinates of the location of the structure. (number of values = numCoordinates)",
+            alias="yCoordinates",
+        )
+
+    comments: Comments = Field(default_factory=Comments)
+
+    header: Literal["Structure"] = "Structure"
+
+    id: str = Field("id", max_length=256)
+    name: str = Field("id")
+
+    # TODO: Note that either branch_id and chainage needs to be defined, or the coordinates. This shoud be either validated or refactored.
+    branch_id: Optional[str] = Field(None, alias="branchId")
+    chainage: Optional[float] = None
+
+    n_coordinates: Optional[int] = Field(None, alias="numCoordinates")
+    x_coordinates: Optional[List[float]] = Field(None, alias="xCoordinates")
+    y_coordinates: Optional[List[float]] = Field(None, alias="yCoordinates")
+
+
+class FlowDirection(str, Enum):
+    none = "none"
+    positive = "positive"
+    negative = "negative"
+    both = "both"
+
+
+class Weir(Structure):
+    class Comments(Structure.Comments):
```

Comment:
That's some odd inheritance, but I guess it otherwise wouldn't actually inherit your abc one?

Reply:
Might we also move the class comments to underneath the normal fields? Ah, that would break the comments field itself I guess.

Reply:
Yeah unfortunately, the `Comments` need to be defined before the field. 
I wanted to make sure that the `Weir.Comments` still has the fields of `Structure.Comments`. I do agree it looks a bit odd, but I'm not sure if there is a better way to do this. We could add the additional fields through some class method which would be called in the IniBasedModel, and then the fields would be added at that point. However, if my understanding of pylance and other ide intellisense is correct, it would be hard to automatically retrieve the fields (and mypy / pylance would probably complain). 

I personally would say that the additional type hinting you get by this inheritance trumps how it looks, seen as much as the help with the fields by end users. But I might be wrong there.